sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RemoveRedundantAggregates.scala

```diff
@@ -47,6 +47,17 @@ object RemoveRedundantAggregates extends Rule[LogicalPlan] with AliasHelper {
       } else {
         newAggregate
       }
+
+     case agg @ Aggregate(groupingExps, _, child)
+         if agg.groupOnly && child.deterministic &&
+           child.distinctKeys.exists(_.subsetOf(ExpressionSet(groupingExps))) =>
+      Project(agg.aggregateExpressions, child)
+
+    case agg @ Aggregate(groupingExps, aggregateExps, child)
+        if aggregateExps.forall(a => a.isInstanceOf[Alias] && a.children.forall(_.foldable)) &&
```

Comment:
how about a mix? e.g. `SELECT a, 1, b FROM ... GROUP BY a, b, c`

Reply:
We already support this case, add a new test case:
https://github.com/apache/spark/pull/35779/files#diff-7cd933ffc7b9ce86d5973bee80f4a5bd4a021c0f0ff81defe1f020bcb55b4b3bR153-R159